dojo/tools/trufflehog/parser.py

```diff
@@ -69,7 +69,10 @@ def __init__(self, filename, test):
 
     def parse_json(self, json_output):
         try:
-            json_data = json.loads(json_output)
+            try:
+                json_data = json.loads(str(json_output, 'utf-8'))
```

Comment:
What will happen for those bytes that are not utf-8?
If you have a doubt you're really reading utf-8, I believe the clean way is : 
`json_output.decode('utf-8')`
by defaut this will throw an exception if it's not utf-8, but you can also : 
`json_output.decode('utf-8', errors='replace')`
which will replace unknown characters by �
Or
`json_output.decode('utf-8', errors='ignore')`
which will strip out unknown characters

From that point on you're working with a <type 'bytes'> (json_output was a <type 'str'>)
when you're finished working with it and want a type str (to pass to the database for ex) then you'll need to 
`encode('utf8')`

I believe you need to encode before passing to the loads function as in version 3.5, `loads` doesn't support bytes yet: 
https://docs.python.org/3/library/json.html
```
json.loads(s, *, encoding=None, cls=None, object_hook=None, parse_float=None, parse_int=None, parse_constant=None, object_pairs_hook=None, **kw)
Deserialize s (a str, bytes or bytearray instance containing a JSON document) to a Python object using this conversion table.
(...)
Changed in version 3.6: s can now be of type bytes or bytearray. The input encoding should be UTF-8, UTF-16 or UTF-32.
```

so in the end this should be fine with for example: (don't trust me though please test :) I don't see unit tests and don't have samples with me...)
`json_data = json_output.decode('utf-8', errors='replace').encode('utf-8')`

(no exception handling needed)

Reply:
@ptrovatelli Good point on the chars that are not utf-8. I had not thought about that. The reason for the exception handling is that the files are not always bytes or strings. For example, I tried a fix that handled imports correctly, but would fail the unit tests. They used the same function, so I was totally stumped. Unfortunately, there's about 25 other parsers that suffered from this issue. Here is the rest on #1467. I had forgotten Trufflehog in the cluster!

Reply:
OK. it's just that json.load takes utf-8 strings in input so it seems strange to give a string that failed str(..., utf-8). However after a few tests it doesn't seem as easy as I expected, i'll need to dig it a little to see what's what.

Reply:
Believe me, I know. It nearly took me three times to get it working! I've seen the import through GUI/API reads files as bytes. I suspect it's because of the temp file upload. The unit tests open the file to read so it auto converts to string. Thats why I did it in that order.